### PR TITLE
 Optimize Error/Loading States and Prevent Re-renders

### PR DESCRIPTION
**Pull request by tanya-newStart**
- Replaced inline error/loading text with dedicated components
- Implemented proper loading states for add buttons
- Optimized component re-renders

### CODE REVIEW

File: app/app/components/AddBookToBookshelf/BookGrid.jsx [hunk 1]
```diff
@@ -1,39 +1,72 @@
...
-const BookGrid = ({ books, onAddBook, loading }) => {
+const BookCard = memo(({ book, onAddBook }) => {
+  const [isAdding, setIsAdding] = useState(false);
+  const isMounted = useRef(true);
+
+  useEffect(() => {
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
```
Comment (magdazelena) [inline app/app/components/AddBookToBookshelf/BookGrid.jsx:13]: why is this needed ?

**Review by magdazelena — COMMENTED** [review]
great work!

File: app/app/components/AddBookToBookshelf/BookGrid.jsx (hunk 1 above)
Comment (tanya-newStart) [reply app/app/components/AddBookToBookshelf/BookGrid.jsx:13]: I think I was trying to fix bad state call

Comment (magdazelena) [reply app/app/components/AddBookToBookshelf/BookGrid.jsx:13]: seems a little convoluted... It's probably not enough time to fix things nor it is necessary, but "bad state call" usually means something can be simplified in the component :) This is things usually take a long time to debug, so I understand and applaud the solution 👏
